templateManager.ts

```diff
@@ -140,9 +135,17 @@ export class TemplateManager {
                 wsUrl.protocol = wsUrl.protocol == 'https:' ? 'wss:' : 'ws:';
                 let ws = new WebSocket(wsUrl);
 
-                ws.addEventListener('open', (_) => {
+                ws.addEventListener('open', async (_) => {
                     console.log(`successfully connected to websocket for ${serverUrl}`);
                     this.websockets.push(ws);
+                    if (isTopLevelTemplate) {
```

Comment:
why is this in ws.open?

Reply:
You think it's better in the spot where we first get the topics?

Reply:
yeah, right after L131 this.notificationTypes.set(domain, topics);

Reply:
Ah, it needs to be in an async function because tampermonkey is weird (GM.setValue only works correctly in async context for some reason) so the entire setupNotifications() would need to be made async.

Reply:
> Ah, it needs to be in an async function because tampermonkey is weird (GM.setValue only works correctly in async context for some reason) so the entire setupNotifications() would need to be made async.

that's fine i think, it doesn't return anything

Reply:
also, now that .forced is back, you could just do it here https://github.com/osuplace/templateManager/pull/23/commits/9f73507e318465caf5be4725882ae192aaa03f82#diff-f68cb8c967655365ad48f2337f5ca25ed4978381fad4ee02c2ccf581d0f47e39R220